A30-xds-v3.md

```diff
@@ -0,0 +1,255 @@
+  `/envoy.service.discovery.v3.AggregatedDiscoveryService/StreamAggregatedResources`.
+  We call this the *transport protocol version*.
+- The type of resources that we request via the transport protocol.  For
+  example, the type for a Listener resource in v2 is `envoy.api.v2.Listener`,
+  whereas in v3 the type is `envoy.config.listener.v3.Listener`.  We
+  call this the *resource version*.
+
+When gRPC uses v3 to talk to a server, the primary thing that it is
+choosing is the transport protocol version, not the resource version.
+
+Luckily, we can be more flexible with the resource version, because
+there's essentially only one case where we use a field in the v2 protos
+that was removed in the v3 protos.  (That case is the node `build_version`
+field, which we can handle as a special case.)  Therefore, we can
+simply change our code to parse all incoming messages as v3, regardless
+of whether they are actually v2 or v3 messages.  We will also need to
+accept both v2 and v3 type names interchangeably in all `type_url` fields.
+
+This means that the logic for determining the transport protocol version
+is completely independent of the logic for handling the resource version.
+gRPC will always request resources of the same version as the transport
+protocol it is using.  However, it will accept any supported version of
+resources in responses from the server, regardless of what version of
+the transport protocol we are using.
+
+In other words, if we're using the v2 transport protocol, we will
+request v2 resources, but it's fine for the server to send us either v2
+or v3 resources.  Similarly, if we're using the v3 transport protocol,
+we will request v3 resources, but it's fine for the server to send us
+either v2 or v3 resources.
+
+### Version of LRS Protocol
+
+For now, we will use the same server feature in the bootstrap file to
+indicate both the xDS and the LRS transport protocol version to use.
+
+Note that this means that there will not be a way to configure the client
+to use different transport versions for the two protocols.  In the future,
+if this becomes a problem, we can make use of `transport_api_version`
+field being added in https://github.com/envoyproxy/envoy/pull/11754 to
+specify the transport version of LRS.  However, since this field will exist
+only in xDS v3, this would still not provide a clean way to specify use
+of LRS v3 while using xDS v2.
+
+Just as with xDS, the LRS client will parse responses as v3, regardless
+of which version of the transport protocol is being used.
+
+### Environment Variable
+
+Initially, support for xDS v3 in gRPC will be considered experimental,
+since we are planning to release this support in the client before we
+have been able to perform interop testing with any real-world v3 xDS
+server.  As a result, we are taking the added safety precaution that the
+new server feature in the bootstrap file will only be used if the
+`GRPC_XDS_EXPERIMENTAL_V3_SUPPORT` environment variable is set to
```

Comment:
This is a good idea and it will be nice to standardize the mapping from the server_feature value to the env variable name. So for server_feature of "xds_v3" the env var name to check is "GRPC_EXPERIMENTAL_SUPPORT_XDS_V3"  i.e. you append capitalized "XDS_V3" to "GRPC_EXPERIMENTAL_SUPPORT_"